camel/agents/chat_agent.py

```diff
@@ -234,45 +278,88 @@ def step(
                 and information about the chat session.
         """
         messages = self.update_messages('user', input_message)
-        if self.message_window_size is not None and len(
-                messages) > self.message_window_size:
-            messages = [ChatRecord('system', self.system_message)
-                        ] + messages[-self.message_window_size:]
-        openai_messages = [record.to_openai_message() for record in messages]
-        num_tokens = num_tokens_from_messages(openai_messages, self.model)
 
-        output_messages: Optional[List[BaseMessage]]
+        output_messages: List[BaseMessage]
         info: Dict[str, Any]
-
-        if num_tokens < self.model_token_limit:
+        called_funcs: List[FuncRecord] = []
+        while True:
+            # Format messages and get the token number
+            openai_messages: Optional[List[OpenAIMessage]]
+            num_tokens: int
+            openai_messages, num_tokens = self.truc_messages(messages)
+
+            # Terminate when number of tokens exceeds the limit
+            if openai_messages is None:
+                return self.step_token_exceed(num_tokens, called_funcs)
+
+            # Obtain LLM's response and validate it
             response = self.model_backend.run(openai_messages)
             self.validate_model_response(response)
+
             if not self.model_backend.stream:
+                # flake8: noqa :E501
```

Comment:
emm this one is to avoid PEP8 reporting error due to that this line is too long. I think the first line of this statement includes 4 returned values and it is not good to cut it off in the middle.

Reply:
Haha. Please follow the rule of 80 symbols. Do not supress the warnings like this. If needed, split into several lines with brackets.